### PR TITLE
Implement native camera on iOS

### PR DESCRIPTION
**Pull request by 3nids**

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 45dee682a5cfab35e501068c480daff810fcb0e9:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2143-1ca94ed9d8576fa1d01c048486fc8e7f47d73e91-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2143-1ca94ed9d8576fa1d01c048486fc8e7f47d73e91-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2143-1ca94ed9d8576fa1d01c048486fc8e7f47d73e91-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2143-1ca94ed9d8576fa1d01c048486fc8e7f47d73e91-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for be19c1ba66d05aa3f68846ff43a40eb7c8397802:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2143-2d39e9dae9a9614590389cba3a14db4804e509e3-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2143-2d39e9dae9a9614590389cba3a14db4804e509e3-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2143-2d39e9dae9a9614590389cba3a14db4804e509e3-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2143-2d39e9dae9a9614590389cba3a14db4804e509e3-x86.apk)

**Comment by 3nids** [comment]
ready for review

**Comment by nirvn** [comment]
@3nids , nice work!

File: src/core/platforms/ios/iospicturesource.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/***************************************************************************
+  iospicturesource.h - IosPictureSource
+
+ ---------------------
+ begin                : 5.7.2016
+ copyright            : (C) 2016 by Matthias Kuhn
```
Comment (m-kuhn) [inline src/core/platforms/ios/iospicturesource.h:6]: Looks trustworthy!

File: src/core/qfieldcloudprojectsmodel.cpp [hunk 2]
```diff
@@ -772,6 +772,7 @@ void QFieldCloudProjectsModel::uploadProject( const QString &projectId, const bo
   const QFileInfo projectInfo( QFieldCloudUtils::localProjectFilePath( mUsername, projectId ) );
   const QDir projectDir( projectInfo.absolutePath() );
   QStringList attachmentFileNames = deltaFileWrapper->attachmentFileNames().keys();
```
Comment (m-kuhn) [inline src/core/qfieldcloudprojectsmodel.cpp:774]:
```suggestion
  const QStringList attachmentFileNames = deltaFileWrapper->attachmentFileNames().keys();
```

**Review by m-kuhn — APPROVED** [review]
Can you check indentation for .mm files?

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 6437d29f01c93629a2ed20edc4e16030646ac864:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2143-8156507622d54877a90a70cfa0b2d5f8045aeb46-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2143-8156507622d54877a90a70cfa0b2d5f8045aeb46-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2143-8156507622d54877a90a70cfa0b2d5f8045aeb46-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2143-8156507622d54877a90a70cfa0b2d5f8045aeb46-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for d6fec69a3fb1bedaf80638df705529dd33801fb3:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2143-43494eac14849fa97faf660fcef7b21d6d0831c3-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2143-43494eac14849fa97faf660fcef7b21d6d0831c3-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2143-43494eac14849fa97faf660fcef7b21d6d0831c3-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2143-43494eac14849fa97faf660fcef7b21d6d0831c3-x86.apk)

**Comment by 3nids** [comment]
@m-kuhn I've dropped the const addition you made since it was breaking the build. I'll do a separate PR with your change

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 28b865349cc7c682963a3baa8c7fbc597339dad6:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2143-0ed9f813fed61b5509d77b0f9cafb21a040baae6-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2143-0ed9f813fed61b5509d77b0f9cafb21a040baae6-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2143-0ed9f813fed61b5509d77b0f9cafb21a040baae6-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2143-0ed9f813fed61b5509d77b0f9cafb21a040baae6-x86.apk)

**Pull request merged**